lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -2261,6 +2293,46 @@ def validate_sale_records_response(self, res, course_registration_code, invoice,
         self.assertEqual(res['total_used_codes'], used_codes)
         self.assertEqual(res['total_codes'], 5)
 
+    def test_get_problem_responses(self):
+        """
+        Test whether get_problem_responses returns an appropriate status
+        message when users request a CSV file listing student answers
+        to a given problem.
+        """
+        url = reverse(
+            'get_problem_responses',
+            kwargs={'course_id': unicode(self.course.id)}
+        )
+        problem_url = ''
+
+        # Invalid problem URL
+        response = self.client.get(url, {'problem_url': problem_url})
+        res_json = json.loads(response.content)
+        self.assertEqual(res_json, 'Could not find problem with this URL.')
+
+        # Pretend that problem URL is valid
+        mock_problem_key = Mock(return_value=u'')
+        mock_problem_key.course_key = self.course.id
+        with patch.object(UsageKey, 'from_string') as patched_method:
+            patched_method.return_value = mock_problem_key
+
+            # Successful case:
+            response = self.client.get(url, {'problem_url': problem_url})
+            res_json = json.loads(response.content)
+            self.assertIn('status', res_json)
+            status = res_json['status']
+            self.assertIn('is being created', status)
+            self.assertNotIn('already in progress', status)
+
+            # CSV generation already in progress:
```

Comment:
It feels like you have 4 separate tests in one method. I think you should be more modular.